### PR TITLE
Upgrade Fabric from v1.0.0-rc1 to v1.0.5

### PR DESCRIPTION
**Pull request by antitoine**
This is the PR that will upgrade the tutorial from the v1.0.0-rc1 of Hyperledger Fabric to the v1.0.5. This will answer the issue #1 and fix the certificates expired in the v1.0.0-rc1 (see #2).

### CODE REVIEW

File: blockchain/setup.go [hunk 1]
```diff
@@ -1,210 +1,121 @@
...
-	// Give the organisation user to the client for next proposal
-	client.SetUserContext(orgUser)
+	// Allow orderer to process channel creation
+	time.Sleep(time.Second * 5)
 
-	// Setup Event Hub
-	// This will allow us to listen for some event from the chaincode
-	// and act on it. We won't use it for now.
-	eventHub, err := getEventHub(client)
+	// Org resource management client
```
Comment (antitoine) [inline blockchain/setup.go]: Explain in more detail.

File: blockchain/setup.go [hunk 2]
```diff
@@ -1,210 +1,121 @@
...
-	}
-	if err := eventHub.Connect(); err != nil {
-		return nil, fmt.Errorf("Failed eventHub.Connect() [%s]", err)
+		return fmt.Errorf("failed to create new resource management client: %v", err)
 	}
-	setup.EventHub = eventHub
 
-	// Tell that the initialization is done
-	setup.Initialized = true
+	// Org peers join channel
```
Comment (antitoine) [inline blockchain/setup.go]: Explain in more detail. What happen here?

File: blockchain/setup.go [hunk 3]
```diff
@@ -1,210 +1,121 @@
...
-	// The authentication will be made with local certificates
-	orgUser, err := fcutil.GetPreEnrolledUser(
-		client,
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/keystore",
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/signcerts",
-		"peerorg1Admin",
-	)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to get the organisation user failed: %v", err)
+	// Create channel
```
Comment (antitoine) [inline blockchain/setup.go]: Explain in more detail. A quick overview of a channel ?

File: blockchain/setup.go [hunk 4]
```diff
@@ -1,210 +1,121 @@
...
-func getEventHub(client api.FabricClient) (api.EventHub, error) {
-	eventHub, err := events.NewEventHub(client)
-	if err != nil {
-		return nil, fmt.Errorf("Error creating new event hub: %v", err)
-	}
-	foundEventHub := false
-	peerConfig, err := client.GetConfig().GetPeersConfig()
+func (setup *FabricSetup) InstallAndInstantiateCC() error {
+
+	// Create chaincode package for our chaincode
```
Comment (antitoine) [inline blockchain/setup.go]: Change comment

File: blockchain/setup.go [hunk 5]
```diff
@@ -1,210 +1,121 @@
...
-				p.TLS.Certificate, p.TLS.ServerHostOverride)
-			foundEventHub = true
-			break
-		}
+		return fmt.Errorf("failed to create chaincode package: %v", err)
 	}
 
-	if !foundEventHub {
-		return nil, fmt.Errorf("No EventHub configuration found")
+	// Install our chaincode on org peers
```
Comment (antitoine) [inline blockchain/setup.go]: Explain in more detail. Tell what is the install process (regarding instantiate process).

File: blockchain/setup.go [hunk 6]
```diff
@@ -1,210 +1,121 @@
...
+	installCCReq := resmgmt.InstallCCRequest{Name: setup.ChainCodeID, Path: setup.ChaincodePath, Version: "1.0", Package: ccPkg}
+	_, err = setup.admin.InstallCC(installCCReq)
+	if err != nil {
+		return fmt.Errorf("failed to install cc to org peers %v", err)
 	}
 
-	return eventHub, nil
-}
+	// Set up chaincode policy
```
Comment (antitoine) [inline blockchain/setup.go]: Explain in more detail. What is a chaincode policy ? What will happen in this case ?

File: blockchain/setup.go [hunk 7]
```diff
@@ -1,210 +1,121 @@
...
-		setup.Client, // The SDK client
-		setup.Channel, // The channel concerned
-		setup.ChaincodeId,
-		setup.ChaincodePath,
-		setup.ChaincodeVersion,
-		nil,
-		setup.Channel.GetPeers(), // Peers concerned by this change in the channel
-		setup.ChaincodeGoPath,
-	)
+	// Org resource manager will instantiate our chaincode on the channel
```
Comment (antitoine) [inline blockchain/setup.go]: Explain in more detail. Tell what is the instantiate process (regarding install process).

File: blockchain/setup.go [hunk 8]
```diff
@@ -1,210 +1,121 @@
...
+	setup.client, err = setup.sdk.NewClient(fabsdk.WithUser("User1")).Channel(setup.ChannelID)
 	if err != nil {
-		return err
-	} else {
-		fmt.Printf("Chaincode %s instantiated (version %s)\n", setup.ChaincodeId, setup.ChaincodeVersion)
+		return fmt.Errorf("failed to create new channel client: %v", err)
 	}
 
+	// Clean the client
+	defer setup.client.Close()
```
Comment (antitoine) [inline blockchain/setup.go]: You can't put this here, you need to have a Close function in the Setup struct that the main process will call at the end (with a `defer`).

File: config.yaml [hunk 9]
```diff
@@ -1,51 +1,121 @@
...
+  eventService:
+    timeout:
+      connection: 3s
+      registrationResponse: 3s
+  orderer:
+    timeout:
+      connection: 3s
+      response: 5s
+
+  cryptoconfig:
```
Comment (antitoine) [inline config.yaml]: Comment ?

File: config.yaml [hunk 10]
```diff
@@ -1,51 +1,121 @@
...
+    path: "${GOPATH}/src/github.com/chainHero/heroes-service/fixtures/crypto-config"
+
+  credentialStore:
```
Comment (antitoine) [inline config.yaml]: Comment ?

File: config.yaml [hunk 11]
```diff
@@ -1,51 +1,121 @@
...
+    path: "/tmp/heroes-service-kvs"
+    cryptoStore:
+      path: "/tmp/heroes-service-msp"
+
+   # BCCSP config for the client. Used by GO SDK.
+  BCCSP:
```
Comment (antitoine) [inline config.yaml]: Comment ? What is a BCCSP ?

File: config.yaml [hunk 12]
```diff
@@ -1,51 +1,121 @@
...
+  tlsCerts:
+    systemCertPool: false
+
+channels:
+  chainhero:
+    orderers:
+      - orderer.hf.chainhero.io
+    peers:
+      peer0.org1.hf.chainhero.io:
+        endorsingPeer: true
```
Comment (antitoine) [inline config.yaml]: Comments ?

File: config.yaml [hunk 13]
```diff
@@ -1,51 +1,121 @@
...
+        ledgerQuery: true
+        eventSource: true
+      peer1.org1.hf.chainhero.io:
+        endorsingPeer: true
+        chaincodeQuery: true
+        ledgerQuery: true
+        eventSource: true
+
+    chaincodes:
+      # the format follows the "canonical name" of chaincodes by fabric code
```
Comment (antitoine) [inline config.yaml]: Comments ?

File: config.yaml [hunk 14]
```diff
@@ -1,51 +1,121 @@
...
+    peers:
+      - peer0.org1.hf.chainhero.io
+      - peer1.org1.hf.chainhero.io
+    certificateAuthorities:
+      - ca.org1.hf.chainhero.io
+
+# List of orderers to send transaction and channel create/update requests to. For the time being only one orderer is needed.
+orderers:
+  orderer.hf.chainhero.io:
+    url: grpcs://localhost:7050
```
Comment (antitoine) [inline config.yaml]: Comments ?

File: config.yaml [hunk 15]
```diff
@@ -1,51 +1,121 @@
...
+    grpcOptions:
+      ssl-target-name-override: orderer.hf.chainhero.io
+      grpc-max-send-message-length: 15
+    tlsCACerts:
+      path: "${GOPATH}/src/github.com/chainHero/heroes-service/fixtures/crypto-config/ordererOrganizations/hf.chainhero.io/tlsca/tlsca.hf.chainhero.io-cert.pem"
+
+# List of peers to send various requests to, including endorsement, query and event listener registration.
+peers:
+  peer0.org1.hf.chainhero.io:
+    url: grpcs://localhost:7051
```
Comment (antitoine) [inline config.yaml]: Comments ?

File: config.yaml [hunk 16]
```diff
@@ -1,51 +1,121 @@
...
+    eventUrl: grpcs://localhost:8053
+    grpcOptions:
+      ssl-target-name-override: peer1.org1.hf.chainhero.io
+      grpc.http2.keepalive_time: 15
+    tlsCACerts:
+      path: "${GOPATH}/src/github.com/chainHero/heroes-service/fixtures/crypto-config/peerOrganizations/org1.hf.chainhero.io/tlsca/tlsca.org1.hf.chainhero.io-cert.pem"
+
+certificateAuthorities:
+  ca.org1.hf.chainhero.io:
+    url: https://localhost:7054
```
Comment (antitoine) [inline config.yaml]: Comments ?

File: config.yaml [hunk 17]
```diff
@@ -1,51 +1,121 @@
...
+    httpOptions:
+      verify: false
+    registrar:
+      enrollId: admin
```
Comment (antitoine) [inline config.yaml]: Comments ?

File: install.sh [hunk 18]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
```
Comment (antitoine) [inline install.sh]: Not needed for this tutorial. Keep it if you want to, but remove this script from this repository.

File: main.go [hunk 19]
```diff
@@ -2,55 +2,43 @@ package main
...
-
 func main() {
-	// Setup correctly the GOPATH in the environment
-	if goPath := os.Getenv("GOPATH"); goPath == "" {
-		os.Setenv("GOPATH", defaultGOPATH())
+	// Definition of the Fabric SDK properties
+	fSetup := blockchain.FabricSetup{
+		// Channel parameters
+		ChannelID:     "chainhero",
+		ChannelConfig: "" + os.Getenv("GOPATH") + "/src/github.com/chainHero/heroes-service/fixtures/artifacts/",
```
Comment (antitoine) [inline main.go]: Set the full path here and remove the prefix `"" + ` (no required): `os.Getenv("GOPATH") + "/src/github.com/chainHero/heroes-service/fixtures/artifacts/chainhero.channel.tx"`

File: blockchain/setup.go [hunk 20]
```diff
@@ -1,210 +1,121 @@
...
-		setup.ChaincodeId,
-		setup.ChannelId,
-		[]string{"init"}, // Arguments for the invoke request
-		setup.ChaincodePath,
-		setup.ChaincodeVersion,
-		[]api.Peer{setup.Channel.GetPrimaryPeer()}, // Which peer to contact
-		setup.EventHub,
-	)
+	// Channel client is used to query and execute transactions
+	setup.client, err = setup.sdk.NewClient(fabsdk.WithUser("User1")).Channel(setup.ChannelID)
```
Comment (antitoine) [inline blockchain/setup.go]: Move the "User1" in configuration of the Setup struct

File: blockchain/setup.go [hunk 21]
```diff
@@ -1,210 +1,121 @@
...
-	orgUser, err := fcutil.GetPreEnrolledUser(
-		client,
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/keystore",
-		"peerOrganizations/org1.example.com/users/Admin@org1.example.com/signcerts",
-		"peerorg1Admin",
-	)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to get the organisation user failed: %v", err)
+	// Create channel
+	req := chmgmt.SaveChannelRequest{ChannelID: setup.ChannelID, ChannelConfig: setup.ChannelConfig + "chainhero.channel.tx", SigningIdentity: orgAdminUser}
```
Comment (antitoine) [inline blockchain/setup.go]: See comment in the main.go to remove the "chainhero.channel.tx" here

File: blockchain/setup.go (hunk 1 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 2 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 3 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 4 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 5 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 6 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 7 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 8 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: config.yaml (hunk 9 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: config.yaml (hunk 10 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: config.yaml (hunk 11 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: config.yaml (hunk 12 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: config.yaml (hunk 13 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: config.yaml (hunk 14 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: config.yaml (hunk 16 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: config.yaml (hunk 17 above)
Comment (antitoine) [reply config.yaml]: @MaximeAubanel

File: install.sh (hunk 18 above)
Comment (antitoine) [reply install.sh]: @MaximeAubanel

File: main.go (hunk 19 above)
Comment (antitoine) [reply main.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 20 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 21 above)
Comment (antitoine) [reply blockchain/setup.go]: @MaximeAubanel

File: blockchain/setup.go (hunk 2 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: blockchain/setup.go (hunk 3 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: blockchain/setup.go (hunk 4 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: blockchain/setup.go (hunk 6 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: blockchain/setup.go (hunk 5 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: blockchain/setup.go (hunk 7 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: blockchain/setup.go (hunk 8 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: config.yaml (hunk 9 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 10 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 11 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 12 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 13 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 14 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 15 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 16 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: config.yaml (hunk 17 above)
Comment (MaximeAubanel) [reply config.yaml]: Done

File: install.sh (hunk 18 above)
Comment (MaximeAubanel) [reply install.sh]: Done

File: main.go (hunk 19 above)
Comment (MaximeAubanel) [reply main.go]: Done

File: blockchain/setup.go (hunk 20 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

File: blockchain/setup.go (hunk 21 above)
Comment (MaximeAubanel) [reply blockchain/setup.go]: Done

**Pull request merged**